parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -317,42 +336,70 @@ pub mod pallet {
 			let encoded =
 				enqueued_message.encode().try_into().map_err(|_| SubmitError::MessageTooLarge)?;
 
-			let ticket =
-				OutboundQueueTicket { id: message_id, origin: message.origin, message: encoded };
+			let priority = match message.command {
+				Command::Upgrade { .. } | Command::SetOperatingMode { .. } => Priority::High,
+				_ => Priority::Low,
+			};
+
+			let ticket = OutboundQueueTicket {
+				id: message_id,
+				origin: message.origin,
+				message: encoded,
+				priority,
+			};
 
 			Ok((ticket, delivery_fee))
 		}
 
 		fn submit(ticket: Self::Ticket) -> Result<MessageHash, SubmitError> {
-			// Make sure the bridge not halted
-			Self::ensure_not_halted().map_err(|_| SubmitError::BridgeHalted)?;
-			T::MessageQueue::enqueue_message(
-				ticket.message.as_bounded_slice(),
-				AggregateMessageOrigin::Parachain(ticket.origin),
-			);
+			let from_self = ticket.origin == T::OwnParaId::get();
+			let origin: AggregateMessageOrigin;
+			if from_self {
+				if ticket.priority == Priority::High {
+					origin = AggregateMessageOrigin::SelfChain(Priority::High);
+					// Increase PendingHighPriorityMessageCount by one
+					PendingHighPriorityMessageCount::<T>::mutate(|count| {
+						*count = count.saturating_add(1)
+					});
+				} else {
+					origin = AggregateMessageOrigin::SelfChain(Priority::Low);
+					Self::ensure_not_halted().map_err(|_| SubmitError::BridgeHalted)?;
+				}
+			} else {
+				origin = AggregateMessageOrigin::Parachain(ticket.origin);
+				Self::ensure_not_halted().map_err(|_| SubmitError::BridgeHalted)?;
+			}
+			T::MessageQueue::enqueue_message(ticket.message.as_bounded_slice(), origin);
 			Self::deposit_event(Event::MessageQueued { id: ticket.id });
 			Ok(ticket.id)
 		}
 	}
 
 	impl<T: Config> ProcessMessage for Pallet<T> {
 		type Origin = AggregateMessageOrigin;
+		#[allow(clippy::clone_double_ref)]
```

Comment:
We should fix this. Do you know what the error message is so I can help with it?

Reply:
https://github.com/Snowfork/snowbridge/pull/949/commits/cfb1cfba8659f4b459bb68f0d6c1c7959bb06b84